internal/consensus/reactor.go

```diff
@@ -561,31 +560,10 @@ func (conR *Reactor) sendNewRoundStepMessage(peer p2p.Peer) {
 	})
 }
 
-func (conR *Reactor) updateRoundStateRoutine() {
-	t := time.NewTicker(100 * time.Microsecond)
-	defer t.Stop()
-	for range t.C {
-		if !conR.IsRunning() {
-			return
-		}
-		rs := conR.conS.GetRoundState()
-		conR.rsMtx.Lock()
-		conR.rs = rs
-		conR.rsMtx.Unlock()
-	}
-}
```

Comment:
This loops does very frequent and short lived allocations.
Can it be replaced with just a get over round state wherever is necessary?
This is what this PR tries and do